routers/web/repo/issue.go

```diff
@@ -2413,6 +2413,15 @@ func UpdatePullReviewRequest(ctx *context.Context) {
 			ctx.Status(http.StatusForbidden)
 			return
 		}
+		if issue.PullRequest.HasMerged {
+			log.Warn(
+				"UpdatePullReviewRequest: cannot add or remove a review for merged PR %-v#%d",
+				issue.Repo.FullName(), issue.PullRequest.Index,
+			)
```

Comment:
Overall LGTM, I think it doesn't need to output any log, since it is an expected behavior.

As a quick fix, I could vote my approval.

To have a complete solution, I have some questions:

1. Is it possible to add some tests?
2. Is there anything need to change for related API? (if some related API endpoints exist)

Reply:
Thanks for the review @wxiaoguang! I have been working on this throughout the day and think my most recent changes handles the "re-review" and "dismiss" actions, though the quality is questionable (forgive me rusty Go code!).

> Is it possible to add some tests?

I think this would be good too. I'll take a look at how unit testing works here and see if I can make some unit tests.

> Is there anything need to change for related API? (if some related API endpoints exist)

This was something I was thinking about too. When I get the time, I'll see what I can do.

Also, with my current changes, it doesn't account for teams, so I believe they will have the same weird behavior (haven't checked yet). So I might need to handle this in this PR too

Reply:
I also did not consider handle closed PRs, but it's up to you if you want me to tackle that too

Reply:
It's better to handle "closed PRs" together, since the new code seems well structured.

Maybe `IsErrReReviewRequestOnMergedPR` -> `IsErrReReviewOnClosedPR` ? IMO "merged" also means "closed"